contracts/Strategy.sol

```diff
@@ -432,14 +454,24 @@ contract Strategy is BaseStrategy {
         return false;
     }
 
-    function whitelistReward(address _rewardToken, SwapSteps memory _steps) public onlyVaultManagers {
-        rewardToken = IERC20(_rewardToken);
-        rewardToken.approve(address(balancerVault), 0);
-        rewardToken.approve(address(balancerVault), max);
-        swapSteps = _steps;
+    // for partnership rewards like TUSD or airdrops
+    function whitelistRewards(address _rewardToken, SwapSteps memory _steps) public isVaultManager {
+        IERC20 token = IERC20(_rewardToken);
```

Comment:
Should check that this is not want, so we dont max approve here

Reply:
https://github.com/yearn/yearn-strategies/issues/247#issuecomment-1078471499

gave more reasons here